core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -74,6 +74,18 @@ private[kafka] object LogValidator extends Logging {
     }
   }
 
+  private[kafka] def validateRecords(records: MemoryRecords): RecordBatch = {
+    // Assume there's only one batch with compressed memory records; otherwise, return InvalidRecordException
+    val batchIterator = records.batches.iterator
+    val batch = batchIterator.next()
```

Comment:
Do you think it's worth calling `hasNext()` first so that we can print a nice exception if the buffer reaches this point empty? I think we have a check for emptiness already in `analyzeAndValidateRecords`, but maybe no harm being a little defensive.